path_rotate_root.go

```diff
@@ -0,0 +1,256 @@
+package azuresecrets
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/vault-plugin-secrets-azure/api"
+
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/hashicorp/go-multierror"
+
+	"github.com/hashicorp/go-uuid"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+func pathRotateRoot(b *azureSecretBackend) *framework.Path {
+	return &framework.Path{
+		Pattern: "rotate-root",
+		Fields: map[string]*framework.FieldSchema{
+			"expiration": {
```

Comment:
I could go either way on this. I thought it made sense here since it only applies to rotating the root credentials, and nothing to do with any other part of the config. The other rotate-roots don't take any input because they don't have expiration dates like Azure does, so I'm not sure that's a good comparison. For what it's worth, the expiration here is an optional field and will default to ~6 months so you don't have to explicitly specify it unless you want a different value.

Reply:
My vote is in `config` with other root related values, so an operator can configure once and not worry about it anymore.

Reply:
Since we already provide the root's client ID and client secret in the config endpoint, having this in the config endpoint as well seems reasonable. I don't think that this value will change once it's set in there, and I'm not sure if operators would set a different expiration value from one rotation to the other. A side benefit of having it in the config is that it can be read back when one is `GET`'ing the config.